test/functional/api/buffer_spec.lua

```diff
@@ -54,6 +60,15 @@ describe('buffer_* functions', function()
       eq({'a', 'b'}, curbuf('get_line_slice', 0, -1, true, false))
       eq({'b'}, curbuf('get_line_slice', 1, -1, true, false))
       eq({'b', 'c'}, curbuf('get_line_slice', -2, -1, true, true))
+      eq({}, curbuf('get_line_slice', 2, 3, false, true))
```

Comment:
I noticed that over-shooting the buffer contents will "wrap around" an unlimited number of times. This is a bit surprising. For example, in this test, the buffer contents are:

```
a
b
c
```

Then, `get_line_slice(1,18, true, true)` returns `['b', 'c']`. @nhynes is this how python works as well?